### PR TITLE
2D Wave solver - Mesh parser, mesh plotter, isoparametric mapping, and sphinx conf.

### PR DESCRIPTION
**Pull request by amanabt**
This is the beginning of the development of the 2D wave equation solver. This PR I will:

1. Write the README following the Quazartech style guidelines.
2. Write a `params` file which will contain the user inputs required to run the program.
3. Storing of 2<sup>nd</sup> order quadrangular mesh.
4. Write a function to display the mesh grid.
5. Write a function to read the 2<sup>nd</sup> order quadrangular `.msh` files.
6. Configure `sphinx` for this project.
6. Write function to map an element of the mesh to
image coordinates, where,
image
image.
This method of mapping is called isoparametric mapping.

### CODE REVIEW

File: 2d_solver/isoparam.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+def isoparam_x(x_nodes, xi, eta):
+    '''
+    This function allows isoparametric mapping of a :math:`2^{nd}` order
```
Comment (Balavarun5) [inline 2d_solver/isoparam.py]: The usage of 'function' can be avoided

File: 2d_solver/msh_parser.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        return True
+    
+    def save_node (tag, x, y, z):
+        nodes.append([x, y])
+        return
+
+    
+    def is_9_node_quadrangle (eletag, eletype,
+                              physgrp, nodes):
+        return eletype == msh_handler.quadrangle_9_node
```
Comment (Balavarun5) [inline 2d_solver/msh_parser.py]: A variable can be returned instead of an expression

File: 2d_solver/msh_parser.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+    def save_element (eletag, eletype,
+                      physgrp, node_tags):
+
```
Comment (Balavarun5) [inline 2d_solver/msh_parser.py]: Documentation is missing

File: 2d_solver/msh_parser.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+            Array of nodes in the mesh. First column and the second column are
+            the :math:`x` and :math:`y` coordinates respectivily.
+            
+    elements : np.ndarray [N_e, 8]
+               Array of elements.
+               
+    xi_LGL  : np.array [N_LGL]
+              LGL points on the :math:`\\xi` axis
+
+    eta_LGL  : np.array [N_LGL]
```
Comment (Balavarun5) [inline 2d_solver/msh_parser.py]: Alignment of the colons

**Comment by mchandra** [comment]
Does this PR contain all of the code required for 2D wave eqn evolution? Or only isometric mapping and A matrix assembly?

**Comment by amanabt** [comment]
A list of what the PR contains is written in the PR description. It does not contain code for A matrix assembly. I thought that it would be better to give a separate PR for A matrix assembly.

Ok, I think that the current title is not the appropriate gist of the PR. I will change it.

File: 2d_solver/msh_parser.py (hunk 2 above)
Comment (amanabt) [reply 2d_solver/msh_parser.py]: these are one line functions. I don't think that we need to complicate them. The name of the function is descriptive enough.

File: 2d_solver/msh_parser.py (hunk 3 above)
Comment (amanabt) [reply 2d_solver/msh_parser.py]: Name of the function is already descriptive.

File: 2d_solver/msh_parser.py (hunk 4 above)
Comment (amanabt) [reply 2d_solver/msh_parser.py]: checked at all other places and modified the documentations accordingly.

File: 2d_solver/isoparam.py (hunk 1 above)
Comment (amanabt) [reply 2d_solver/isoparam.py]: done

**Pull request merged**